File: exp9/fcfs.py
def FCFS(head, sequence):
    if not sequence:
        return 0
    seek_operations = 0
    print(head,end=" ")
    for i in sequence:
        if i!= head:
            print(" ==> ",i, end=" ")
            difference = abs(head - i)
            seek_operations += difference
            head = i
    return seek_operations
if __name__ == "__main__":
    Number_disk = int(input("Enter the number of disks: "))
    if Number_disk > 0:
        head = int(input("Enter initial header position: "))
        while not head in range(Number_disk + 1):
            head = int(input("Please enter valid initial head position: "))
        sequence = []
        sequence = list(map(int, input("Enter the sequence:").split()))
        for i in sequence:
            if i < 0 or i >= Number_disk:
                print("Sequence out of range")
                exit(0)
        seek_operations = FCFS(head, sequence)
        print("\nTotal number of seek operations: ", seek_operations)

File: exp9/fcfs.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

void main(){

    int queue[20],n,head,i,j,k,seek=0,max,diff;
    printf("FCFS disk secdhuling algo\n");
    printf("Enter maximum range of disk: ");
    scanf("%d",&max);
    printf("Enter maxium no. of queue request: ");
    scanf("%d",&n);
    printf("Enter sequence: ");
    for(i=1;i<=n;i++){
        scanf("%d",&queue[i]);
    }
    printf("Initial head position: ");
    scanf("%d",&head);
    queue[0]=head;
    for( j = 0; j <= n-1; j++)
    {
        diff = abs(queue[j+1]-queue[j]);
        seek+=diff;
        printf("\n%d => %d in sek time: %d \n",queue[j],queue[j+1],diff);
    }
    printf("\nTotal head movements: %d",seek);
    
}